File: 387. First Unique Character in a String/Approach04.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> mp;
        for(auto& ch: s){
            mp[ch]++;
        }
        int index = 0;
        for(auto& ch: s){
            if(mp[ch] == 1){
                return index; 
            }
            index++;
        }
        return -1;
    }
};

File: 387. First Unique Character in a String/Approach04.py
from typing import *
import collections

class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = collections.Counter(s)
        index = 0
        for ch in s:
            if(freq[ch] == 1):
                return index
            index += 1 
        return -1